### ISSUE TITLE
Errors in training Sentence QE data with biRNN.sh

### ISSUE THREAD

**Issue by lihongzheng-nlp** [sheffieldnlp/deepQuest#8]
Following your tutorial step by step,  I put the wmt2017 data into the quest/examples/qe-2017, and didn't change any config file,  then run `./train-test-sentQEbRNN.sh --task qe-2017 --source src --target mt --score hter --activation sigmoid --device cuda0 > log-sentQEbRNN-qe-2017.txt 2>&1 &` as mentioned in the tutorial, it gave the errors as follow:
`cp: cannot stat ‘datasets/Dataset_qe-2017_srcmt.pkl’: No such file or directory
cp: cannot stat ‘trained_models/qe-2017_srcmt_EncSent/epoch_is_weights.h5’: No such file or directory`
I was wondering that these **two files are produced during the training process**, or they must be **prepared before the training**?  I don't where is the **datasets** directory.
Would you please help me solve the problem? Thank you very much!

**Comment by fredblain** [comment]
Hi @VictorLi2017,
by default, the 'TASK_NAME' in `config-sentQEbRNN.py` is set to 'qe-2016'. This variable should match with the folder that contains your data, _i.e._ 'qe-2017' in your case. Could you please double check?
Cheers,

**Comment by lihongzheng-nlp** [comment]
Hello @fredblain thanks for your response. This time I only changed data file name 'qe-2017' to 'qe-2016' and didn't modify `config-sentQEbRNN.py` , then retried the command above, but still the same error. 
Should I need to **modify the related config file first** according to the language pairs ( e.g. zh-en) before I start training?

**Comment by fredblain** [comment]
@VictorLi2017, could you please:
- indicate the content of your `quest/examples/qe-2017` folder;
- share your config file here;
- attach your log file.

Thanks.

**Comment by lihongzheng-nlp** [comment]
@fredblain 

- I Downloaded the task1_en-de_training-dev.tar.gz, task1_en-de_test.tar.gz and wmt17_en-de_gold.tar.gz archives from WMT 17 QE task site, and unarchive all the three archives into the folder (just now I changed the name to 'qe-2016'). so the training and dev files are: **dev/train.src, dev/train.mt, dev/train.hter, dev/train.pe, and test.hter**.

- About the config file. I'm not sure whether I need to modify the config first before the training, so I didn't change anything in the repo to train the official WMT data.

- Here're the log files.
[Uploading log-qe-2016_srcmt_EncSent.txt…]()
[Uploading log-qe-2016_srcmt_EncSent-prep.txt…]()
[Uploading log-sentQEbRNN-qe-2016.txt…]()

**Comment by fredblain** [comment]
@VictorLi2017,
your log files are not attached. Regarding the files you should have under `examples/qe-2017`:
- from ` task1_en-de_training-dev`, you get: `{train, dev}.{src,mt,pe,hter}`
- from `task1_en-de_test`, you get: `test.{2016,2017}.{src,mt}`
- from `wmt17_en-de_gold`, you get: `en-de_task1_test.{2016,2017}.{pe,tags,hter}`

What you need under `qe-2017` to train you sentence-level (BiRNN) QE model, is:
- `train.{src,mt,hter}`
- `dev.{src,mt,hter}`
- `test.{src,mt,hter}` (optional)

To use the test set from 2017, you need to provide a `test.hter` file, which you can do by renaming (or linking to) `en-de_task1_test.2017.hter` (see [here](https://sheffieldnlp.github.io/deepQuest/tutorials.html#examples) -- same for using the test set from 2016).

In your config file, set:
- `TASK_NAME = 'qe-2017'`
- `TEXT_FILES = {'train': 'train.', 'val': 'dev.'}` (optional: add `'test': 'test.'`)
- `EVAL_ON_SETS = ['val']` (optional: add `'test'`)

**Comment by fredblain** [comment]
Hi @VictorLi2017,
is your issue running deepQuest now solved?

**Comment by lihongzheng-nlp** [comment]
Hello @fredblain sorry for replying late, because of some busy things. 

I think the issue is a little complicated. I created an virtual environment, and following the official tutorial of deepquest:
`git clone https://github.com/sheffieldnlp/deepQuest.git`
`cd deepQuest`
`conda install theano`
`pip install -r requirements.txt`

theano 1.0.3 was installed. During the installment of requirements, I have problems of installing coco-caption, it always showed as follows:
`Obtaining coco-caption from git+https://github.com/lvapeab/coco-caption.git#egg=coco-caption
  Cloning https://github.com/lvapeab/coco-caption.git to ./src/coco-caption
  Running command git clone -q https://github.com/lvapeab/coco-caption.git /home/hzli/deepQuest-master/quest/src/coco-caption
 fatal: unable to access 'https://github.com/lvapeab/coco-caption.git/': Failed connect to github.com:443; Connection timed out
ERROR: Command "git clone -q https://github.com/lvapeab/coco-caption.git /home/hzli/deepQuest-master/quest/src/coco-caption" failed with error code 128 in None`
I'm not sure whether `coco-caption` is essential to the deepquest toolkit, so I only installed others in the requirements and tried to go on.

After setting the `config.py` under `deepquest/quest` as you mentioned, I ran following command according to the tutorial:
`export KERAS_BACKEND=theano `
`export MKL_THREADING_LAYER=GNU`
`THEANO_FLAGS=device=cuda0 python main.py`
but it failed no matter I used gpu or cpu, as shown in the attached log. And my `config.py` file is also attached.
From the log, it seems that there's something wrong with compiling theano, but I'm not sure the exact reason.
[config.py.txt](https://github.com/sheffieldnlp/deepQuest/files/3169821/config.py.txt)
[training_cpu.log.log](https://github.com/sheffieldnlp/deepQuest/files/3169822/training_cpu.log.log)
[training_gpu.log.log](https://github.com/sheffieldnlp/deepQuest/files/3169823/training_gpu.log.log)

With the same config.py and same above commands, I successfully ran on my colleague's machine.
 
In a word, currently seems there are two problems which results in the failture:

- whether the theano environment is well-installed or not?

- whether `coco-caption` is essential? and if yes, can we copy it to `deepquest/src` and manually install it?  If yes, how to install it?

**Comment by fredblain** [comment]
HI @VictorLi2017,
the `coco-caption` evaluation package is only required to perform evaluation. For QE, we are reporting results using Pearson, MAE and RMSE, and therefore not the scoring functions from this package. From that, it is not essential but if you wish to modify the code base in order to use the scoring functions from `coco`, then it will become a requirement.  

Since you confirm that you managed to run deepQuest with the same options and parameters, but under an different environment, it indicates that there is something wrong with your own environment (and therefore not with deepQuest itself).  I have personally never used deepQuest in a virtual environment, and I am not aware that somebody else has done it.

This issue was open for training errors which were due to a wrong configuration/setup. This is now fixed, therefore I'm closing this issue.
